src/main/java/com/sap/sgs/phosphor/fosstars/data/github/FirstCommit.java

```diff
@@ -61,27 +47,21 @@ protected FirstCommit doUpdate(GitHubProject project, ValueSet values) throws IO
       return something.get();
     }
 
-    GHRepository repository = github.getRepository(project.path());
-    long millis = repository.getCreatedAt().getTime() + DELTA;
-
-    PagedIterator<GHCommit> iterator = repository.queryCommits()
-        .until(millis)
-        .list()
-        .withPageSize(PAGE_SIZE)
-        .iterator();
-
-    List<GHCommit> lastPage = null;
-    while (iterator.hasNext()) {
-      lastPage = iterator.nextPage();
-    }
-
-    if (lastPage == null) {
-      return UnknownValue.of(FIRST_COMMIT_DATE);
+    Value<Date> value = FIRST_COMMIT_DATE.unknown();
+    Optional<GHCommit> firstCommit = gitHubDataFetcher().firstCommitFor(project);
+    if (firstCommit.isPresent()) {
+      value =
+        FIRST_COMMIT_DATE
+              .value(firstCommit.get().getCommitDate());
     }
-
-    Value<Date> value = FIRST_COMMIT_DATE.value(lastPage.get(lastPage.size() - 1).getCommitDate());
-    cache.put(project, value);
-
+    cache.put(project, value, tomorrow());
     return value;
   }
+
+  /**
+   * Get the Singleton instance of {@link GitHubDataFetcher}.
+   */
+  public GitHubDataFetcher gitHubDataFetcher() {
```

Comment:
Please move this method to `AbstractGitHubDataProvider` to avoid duplicate code.